website/src/components/Edit/DataRow.tsx

```diff
@@ -2,39 +2,67 @@ import { type FC } from 'react';
 import { Tooltip } from 'react-tooltip';
 
 import { InputField, type KeyValuePair, type Row } from './InputField.tsx';
+import type { InputFieldOption } from '../../types/config.ts';
 import WarningAmberIcon from '~icons/ic/baseline-warning-amber';
 import DangerousTwoToneIcon from '~icons/ic/twotone-dangerous';
 
 type EditableRowProps = {
     label?: string;
     inputField: string;
     row: Row;
+    /**
+     * Options for this row.
+     * If given, the input field will have a dropdown to choose an option from.
+     */
+    options: InputFieldOption[] | undefined;
     onChange: (editedRow: Row) => void;
 };
 
-export const EditableDataRow: FC<EditableRowProps> = ({ label, inputField, row, onChange }) => {
+export const EditableDataRow: FC<EditableRowProps> = ({ label, inputField, row, onChange, options }) => {
     const colorClassName = row.errors.length > 0 ? 'text-red-600' : row.warnings.length > 0 ? 'text-yellow-600' : '';
 
     const content = `input metadata name: ${inputField}`;
 
     return (
-        <tr className='table-fixed w-full'>
-            <td className={`w-1/4 relative ${colorClassName}`} data-tooltip-id={'field-tooltip' + row.key}>
-                <label htmlFor={row.key}>{`${label ?? row.key}:`}</label>
-                <Tooltip
-                    id={'field-tooltip' + row.key}
-                    place='bottom-start'
-                    content={content}
-                    className='absolute z-50 top-full left-0 mt-1'
-                />
-            </td>
-            <td className='pr-3 text-right '>
-                <ErrorAndWarningIcons row={row} />
-            </td>
-            <td className='w-3/4'>
-                <InputField row={row} onChange={onChange} colorClassName={colorClassName} />
-            </td>
-        </tr>
+        <>
+            <tr className='table-fixed w-full'>
+                <td className={`w-1/4 relative ${colorClassName}`} data-tooltip-id={'field-tooltip' + row.key}>
+                    <label htmlFor={row.key}>{`${label ?? row.key}:`}</label>
+                    <Tooltip
+                        id={'field-tooltip' + row.key}
+                        place='bottom-start'
+                        content={content}
+                        className='absolute z-50 top-full left-0 mt-1'
+                    />
+                </td>
+                <td className='text-right'>
+                    <div className='pr-2 flex flex-row items-center'>
+                        <ErrorAndWarningIcons row={row} />
+                    </div>
+                </td>
+                <td className='w-3/4'>
+                    <InputField row={row} onChange={onChange} colorClassName={colorClassName} options={options} />
+                </td>
```

Comment:
Doesn't have to be in this PR but is there a way to remove the warnings/errors once the user has selected an option from the dropdown? In the case of the options the field should be valid once an option is chosen from the dropdown.

![image](https://github.com/user-attachments/assets/43af26ee-85f8-43fb-b46e-8e5dc0a98a9f)

Reply:
I think this is basically a more general point about warnings and errors: they apply to the old version not the current entered text. I think it would be a bit weird for them to immediately vanish, but I think we could add stuff to make clear they are from the submitted version.